core/src/main/java/com/radixdlt/sync/LocalSyncService.java

```diff
@@ -428,68 +418,137 @@ private boolean isFullySynced(SyncState.SyncingState syncingState) {
+              .sync()
+              .invalidResponsesReceived()
+              .label(new InvalidSyncResponse(InvalidSyncResponseReason.TRANSACTION_COUNT_MISMATCH))
+              .inc();
+          peerControl.banPeer(sender, PEER_BAN_DURATION, "inconsistent transaction count received");
+          yield this.processSync(currentState.clearPendingRequest().removeCandidate(sender));
+        }
+        case InvalidSyncResponseException.UnparseableTransaction exc -> {
+          log.warn(
+              "LocalSync: Received unparseable transaction in sync response {} from {}",
+              syncResponse,
+              sender);
+          metrics
+              .sync()
+              .invalidResponsesReceived()
+              .label(new InvalidSyncResponse(InvalidSyncResponseReason.UNPARSEABLE_TRANSACTION))
+              .inc();
+          peerControl.banPeer(sender, PEER_BAN_DURATION, "unparseable transaction received");
+          yield this.processSync(currentState.clearPendingRequest().removeCandidate(sender));
+        }
+        case InvalidSyncResponseException.ComputedTransactionRootMismatch exc -> {
+          log.warn(
+              "LocalSync: Received mismatched transaction root in sync response {} from {}",
+              syncResponse,
+              sender);
+          metrics
+              .sync()
+              .invalidResponsesReceived()
+              .label(new InvalidSyncResponse(InvalidSyncResponseReason.MISMATCHED_TRANSACTION_ROOT))
+              .inc();
+          peerControl.banPeer(sender, PEER_BAN_DURATION, "mismatched transaction root received");
+          yield this.processSync(currentState.clearPendingRequest().removeCandidate(sender));
+        }
+        case InvalidSyncResponseException.NoQuorumInValidatorSet exc -> {
+          log.warn(
+              "LocalSync: Received insufficient validator set in sync response {} from {}",
+              syncResponse,
+              sender);
+          metrics
+              .sync()
+              .invalidResponsesReceived()
+              .label(new InvalidSyncResponse(InvalidSyncResponseReason.INSUFFICIENT_VALIDATOR_SET))
+              .inc();
+          peerControl.banPeer(sender, PEER_BAN_DURATION, "insufficient validator set received");
+          yield this.processSync(currentState.clearPendingRequest().removeCandidate(sender));
+        }
+        case InvalidSyncResponseException.ValidatorSignatureMismatch exc -> {
+          log.warn(
+              "LocalSync: Received mismatched signatures in sync response {} from {}",
+              syncResponse,
+              sender);
+          metrics
+              .sync()
+              .invalidResponsesReceived()
+              .label(new InvalidSyncResponse(InvalidSyncResponseReason.MISMATCHED_SIGNATURES))
+              .inc();
+          peerControl.banPeer(sender, PEER_BAN_DURATION, "mismatched signatures received");
+          yield this.processSync(currentState.clearPendingRequest().removeCandidate(sender));
```

Comment:
_(note from author)_

The listing above is lengthy and treats every possible problem separately.
In theory, this could be rolled into 2 major groups: "problems that are probably only caused by a random timeout" vs "problems that strongly indicate a malicious/faulty peer, which should be banned".
I tried this approach, but logging and metric-bumping became a bit cumbersome then.
Hence, the explicit handling seen above.